### PR TITLE
apps/speed.c: Wait for generated children

### PR DESCRIPTION
**Pull request by juergenchrist**
In multi-mode, speed fork()s off several children but does not wait for them.
On Linux, this leads to wrong accounting information of getrusage used by
tools to extract running time and page faults.

Wait for every children and check the return code and termination signal.

Signed-off-by: Juergen Christ <jchrist@linux.ibm.com>

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: apps/speed.c [hunk 1]
```diff
@@ -3581,6 +3583,21 @@ static int do_multi(int multi, int size_num)
...
     }
     OPENSSL_free(fds);
+    for (n = 0; n < multi; ++n) {
+        while (wait(&status) == -1) {
+            if (errno != EINTR) {
+                BIO_printf(bio_err, "Waitng for child failed with 0x%x\n",
+                           errno);
+                return 1;
+            }
+        }
```
Comment (paulidale) [inline apps/speed.c:3593]: We'd normally drop this pair of `{}` since it's unnecessary.

**Review by paulidale — APPROVED** [review]

File: apps/speed.c (hunk 1 above)
Comment (juergenchrist) [reply apps/speed.c:3593]: The one from the `while`?  Shall I drop it?

Comment (t8m) [reply apps/speed.c:3593]: Yes, and yes.

Comment (juergenchrist) [reply apps/speed.c]: Done.

**Review by t8m — APPROVED** [review]
@paulidale still OK?

**Comment by paulidale** [comment]
Yes, good.

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**